app/src/main/java/fr/neamar/kiss/pojo/Pojo.java

```diff
@@ -19,4 +27,68 @@
     // How relevant is this record ? The higher, the most probable it will be
     // displayed
     public int relevance = 0;
+
+
+    /**
+     * Map a position in the normalized name to a position in the standard name string
+     *
+     * @param position Position in normalized name
+     * @return Position in non-normalized string
+     */
+    public int mapPosition(int position) {
+        if(this.namePositionMap != null) {
+            if(position < this.namePositionMap.length) {
+                return this.namePositionMap[position];
+            } else {
+                return this.name.length();
+            }
+        } else{
+            // No mapping defined
+            if(position < this.name.length()) {
+                return position;
+            } else {
+                return this.name.length();
+            }
+        }
+    }
+
+    /**
+     * Set the user-displayable name of this container
+     *
+     * When this method a searchable version of the name will be generated for the name and stored
+     * as `nameNormalized`. Additionally a mapping from the positions in the searchable name
+     * to the positions in the displayable name will be stored (as `namePositionMap`).
+     *
+     * @param name User-friendly name of this container
+     */
+    public void setName(String name) {
+        // Set the actual user-friendly name
+        this.name = name;
+
+        if(name != null) {
+            // Normalize name for faster searching
+            Pair<String, int[]> normalized = StringNormalizer.normalizeWithMap(this.name);
+            this.nameNormalized  = normalized.first;
+            this.namePositionMap = normalized.second;
+        }
+    }
+
+    /**
+     * Set which area of the display name should marked as highlighted in the `displayName`
+     * attribute
+     *
+     * The start and end positions should be offsets in the normalized string and will be converted
+     * to their non-normalized positions before they are used.
+     *
+     * @param positionNormalizedStart Highlighting start position in normalized name
+     * @param positionNormalizedEnd   Highlighting end position in normalized name
+     */
+    public void setDisplayNameHighlightRegion(int positionNormalizedStart, int positionNormalizedEnd) {
```

Comment:
in much the same way, and although i'm not a Java expert by any means, i tried to keep POJO method-free to avoid *invokevirtual* calls (and minimize memory consumption once again)